src/vehicle.cpp

```diff
@@ -3694,6 +3697,84 @@ bool vehicle::sufficient_wheel_config() const
     return true;
 }
 
+bool vehicle::handle_potential_theft( player *p, bool check_only, bool prompt )
+{
+    if( p == nullptr ) {
+        return false;
+    }
+    faction *yours;
+    if( p->is_player() ) {
+        yours = g->faction_manager_ptr->get( faction_id( "your_followers" ) );
+    } else {
+        npc *guy = dynamic_cast<npc *>( p );
+        yours = guy->my_fac;
+    }
+    std::vector<npc *> witnesses;
+    for( npc &elem : g->all_npcs() ) {
+        if( rl_dist( elem.pos(), p->pos() ) < MAX_VIEW_DISTANCE && get_owner() && elem.my_fac == owner &&
+            elem.sees( p->pos() ) ) {
+            witnesses.push_back( &elem );
+        }
+    }
+    // the vehicle is yours, thats fine.
+    if( get_owner() == yours ) {
+        return true;
+        // if There is no owner
+        // handle transfer of ownership
+    } else if( !has_owner() ) {
+        set_owner( yours );
+        remove_old_owner();
+        return true;
+        // if there is a marker for having been stolen, but 15 minutes have passed, then officially transfer ownership
+    } else if( witnesses.empty() && get_owner() && get_owner() != yours && get_old_owner() &&
+               get_old_owner() != yours && theft_time && calendar::turn - *theft_time > 15_minutes ) {
+        if( p->is_player() ) {
+            set_owner( yours );
+            // Npcs dont steal vehicles yet... but maybe one day.
+        } else if( p->is_npc() ) {
+            npc *guy = dynamic_cast<npc *>( p );
+            set_owner( guy->my_fac );
```

Comment:
You can merge these two branches (p being a player and p being a NPC) and use `yours` in both cases. See how you have set the value for that variable (it *is* `guy->my_fac`).